frontend/src/components/features/conversation-panel/microagents-modal.tsx

```diff
@@ -0,0 +1,140 @@
+
+                  return (
+                    <div
+                      key={agent.name}
+                      className="rounded-md overflow-hidden"
+                    >
+                      <button
+                        type="button"
+                        onClick={() => toggleAgent(agent.name)}
+                        className="w-full py-3 px-2 text-left flex items-center justify-between hover:bg-gray-700 transition-colors"
+                      >
+                        <div className="flex items-center">
+                          <h3 className="font-bold text-gray-100">
+                            {agent.name}
+                          </h3>
+                        </div>
+                        <div className="flex items-center">
+                          <span className="px-2 py-1 text-xs rounded-full bg-gray-800 mr-2">
+                            {agent.type === "repo" ? "Repository" : "Knowledge"}
+                          </span>
+                          <span className="text-gray-300">
+                            {isExpanded ? (
+                              <ChevronDown size={18} />
+                            ) : (
+                              <ChevronRight size={18} />
+                            )}
+                          </span>
+                        </div>
+                      </button>
+
+                      {isExpanded && (
+                        <div className="px-2 pb-3 pt-1">
+                          {agent.triggers && agent.triggers.length > 0 && (
+                            <div className="mt-2 mb-3">
+                              <h4 className="text-sm font-semibold text-gray-300 mb-2">
+                                {t("MICROAGENTS_MODAL$TRIGGERS", "Triggers")}
+                              </h4>
+                              <div className="flex flex-wrap gap-1">
+                                {agent.triggers.map((trigger) => (
+                                  <span
+                                    key={trigger}
+                                    className="px-2 py-1 text-xs rounded-full bg-blue-900"
+                                  >
+                                    {trigger}
+                                  </span>
+                                ))}
+                              </div>
+                            </div>
+                          )}
+
+                          <div className="mt-2">
+                            <h4 className="text-sm font-semibold text-gray-300 mb-2">
+                              {t("MICROAGENTS_MODAL$CONTENT", "Content")}
+                            </h4>
+                            <div className="text-sm mt-2 p-3 bg-gray-900 rounded-md overflow-auto text-gray-300 max-h-[400px] shadow-inner">
+                              <pre className="whitespace-pre-wrap font-mono text-sm leading-relaxed">
+                                {agent.content}
```

Comment:
@OpenHands please handle the case when `.content` is empty, in those cases, just display `Microagent has no content`